api/control/v1alpha1/types.go

```diff
@@ -245,6 +267,11 @@ type Destination struct {
 	ClusterId string `json:"clusterId,omitempty"`
 }
 
+type BindingStatus struct {
+	ObservedGeneration int64    `json:"observedGeneration"`
+	Errors             []string `json:"errors,omitempty"`
```

Comment:
I see your point about adding errors field in `Binding` status. 
Having said that, the user creates `BindingPolicy`, not `Binding`. 
I assume a typical K8s user would look for the error in the object he created (meaning in `BindingPolicy`).

Reply:
Indeed. My thinking is that we can add propagation back to the BindingPolicy in a follow-on PR.